code/modules/organs/organ.dm

```diff
@@ -121,10 +121,7 @@
 /obj/item/organ/proc/set_dna(var/datum/dna/new_dna)
 	QDEL_NULL(dna)
 	dna = new_dna.Clone()
-	if(!blood_DNA)
-		blood_DNA = list()
-	blood_DNA.Cut()
-	blood_DNA[dna.unique_enzymes] = dna.b_type
+	LAZYSET(blood_DNA, dna.unique_enzymes, dna.b_type)
```

Comment:
The `Cut()` is now missing, meaning old blood DNA will persist, won't it? Should just do `blood_DNA = list((dna.unique_enzymes) = dna.b_type)` or something, no?